lib/src/kbasesearchengine/system/TypeFileStorage.java

```diff
@@ -62,13 +63,16 @@
                 typeToFile.put(searchType, file);
                 searchTypes.put(searchType, new ArrayList<>(types));
                 final CodeAndType cnt = new CodeAndType(types.get(0).getStorageObjectType());
+                final TypeMapping typeMapping = TypeMapping.getBuilder(cnt.storageCode, cnt.storageType)
```

Comment:
I don't understand the purpose of this change. Previously there'd be a mapping from a storage type to one or more search types in a `TypeMapping`. Now they're split up into multiple `TypeMapping`s, but it's the same information.

Reply:
not the same. There are multiple typemappings for each version of the spec. I thought about adding the search object type into the type mapping but it is redundant b/c we always have a default that has that information. Of course this was based on my understanding of how the specs would work, which may be incorrect...

Reply:
The storageType is refering to kbase storage type. Not the search storage type. by spec, I meant the yaml files we use to index

Reply:
Ok yeah, I misread search type as storage type or got it confused in my head somehow.

> by spec, I meant the yaml files we use to index

The type mapping yaml files or the search type yaml files?